### PR TITLE
fix rdkit descriptors

### PR DESCRIPTION
**Pull request by ARY2260**
## Description
fix rdkit descriptors

## Type of change

Please check the option that is related to your PR.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
  - In this case, we recommend to discuss your modification on GitHub issues before creating the PR
- [ ] Documentations (modification for documents)

## Checklist

- [ ] My code follows [the style guidelines of this project](https://deepchem.readthedocs.io/en/latest/development_guide/coding.html)
  - [ ] Run `yapf -i <modified file>` and check no errors (**yapf version must be  0.32.0**)
  - [ ] Run `mypy -p deepchem` and check no errors
  - [ ] Run `flake8 <modified file> --count` and check no errors
  - [ ] Run `python -m doctest <modified file>` and check no errors
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New unit tests pass locally with my changes
- [ ] I have checked my code and corrected any misspellings

### CODE REVIEW

File: deepchem/feat/molecule_featurizers/rdkit_descriptors.py [hunk 1]
```diff
@@ -105,8 +105,20 @@ def __init__(self,
                     continue
                 if self.use_bcut2d is False and desc_name.startswith('BCUT2D_'):
                     continue
+                if desc_name == 'AvgIpc':  # already handled by 'Ipc' during featurization
+                    continue
                 self.reqd_properties[desc_name] = function
         else:
+            if self.ipc_avg is True:
```
Comment (rbharath) [inline deepchem/feat/molecule_featurizers/rdkit_descriptors.py:112]: Can you add a unit test for these new edge cases?

Comment (ARY2260) [reply deepchem/feat/molecule_featurizers/rdkit_descriptors.py:112]: sir I have added the test

**Comment by ARY2260** [comment]
@rbharath PR is ready for review

**Review by rbharath — COMMENTED** [review]
@ARY2260 I'm seeing some failures in the Torch tests about 159 descriptors vs 160 descriptors on multiple models. I think that may be caused by these changes. Can you take a quick look?

Let's separate out the 'gibrat' fix into a separate small PR so we don't have to block it on the rdkit descriptor changes in this PR
